### ISSUE TITLE
Document what the requirements/dependencies are for upgrading EE engine, UCP and DTR

### ISSUE THREAD

**Issue by mbentley** [docker/docker.github.io#4336]
### Problem description
When viewing the documentation for the Docker EE engine, UCP, and DTR, there is no mention of an upgrade order.  For example, if I have the  EE 17.03 engine, UCP 2.1.x, and DTR 2.2.x and I want to upgrade to EE 17.06 engine, UCP 2.2 and DTR 2.3, what order do I need to execute upgrades for each of the three components?  From the documentation, I do not see any references to this:
https://docs.docker.com/engine/installation/linux/docker-ee/rhel/#install-using-the-repository
https://docs.docker.com/datacenter/ucp/2.2/guides/admin/install/upgrade/
https://docs.docker.com/datacenter/dtr/2.3/guides/admin/upgrade/

The only place this is captured is in the [Customer Success Compatibility Matrix](https://success.docker.com/Policies/Compatibility_Matrix) and even for that, I am not able to find any links to it from the docs so I have to know that exists and find it.

And then if I do find it, I need to decode the matrix.  Could we please spell out in plain english which versions of the other components are supported?  For example, if I am looking at the UCP upgrade docs, it should tell me what version(s) of the engine and DTR are supported and if there is a certain order I need to upgrade (upgrade DTR first? engine first?) and then link me to the relevant upgrade docs.  EE is a platform but upgrades like individual components so we need to have that documented and not left up to guessing.

### Problem location
https://docs.docker.com/engine/installation/linux/docker-ee/rhel/#install-using-the-repository
https://docs.docker.com/datacenter/ucp/2.2/guides/admin/install/upgrade/
https://docs.docker.com/datacenter/dtr/2.3/guides/admin/upgrade/

### Project version(s) affected
All EE engine, UCP and DTR upgrade docs

### Suggestions for a fix
Document what is supported and what order the upgrades must occur.

/cc @vsaraswat @banjot @joaofnfernandes @rkiles

**Comment by mbentley** [comment]
Another question that may be semi relevant is the upgrade order of engines participating in a Swarm. Should one upgrade the managers first or the workers?

**Comment by mark-church** [comment]
Interestingly, this was actually specified in previous docs but not current ones to do managers first then workers: https://docs.docker.com/datacenter/ucp/2.0/guides/installation/upgrade/#upgrade-the-first-controller-node

The specific scenario that needs to be spelled out is how a customer can avoid any app downtime while upgrading their cluster. Customer expectations now are that they won't have to take the full cluster offline to do an upgrade. It will require some sort of manual blue/green scenario on the worker nodes, but it's not 100% clear what might break during an upgrade (overlays between old and new version engines, HRM).

**Comment by clemenko** [comment]
@mark-church and @mbentley  does this work : https://success.docker.com/KBase/Upgrade_an_entire_cluster_with_CentOS%2C_Docker_Engine%2C_UCP%2C_and_DTR ???

**Comment by mbentley** [comment]
This should be in the docs.

**Comment by vsaraswat** [comment]
@joaofnfernandes @JimGalasyn can we assign someone from the docs team to take a look at this? From a holistic perspective I wonder if we can create an "Upgrade Docker EE" under the new Docker EE umbrella. That might make more sense than sticking this in UCP or DTR.

**Comment by joaofnfernandes** [comment]
Reposting my answer from Slack

> We’re still cleaning up after the 17.06.1 release, and triaging issues, so we’ll prioritize that issue against others.

> Cross-stack install/upgrade/backup are high priority, but they’re also tricky to do. Our stack has a ton of permutations, so if we simplify the instructions, someone will complain that it doesn’t work for them. If we tell the whole story, someone will complain that it’s too difficult and create their own docs on the side

**Comment by mbentley** [comment]
I just want upgrade instructions from the previous release (17.03 -> 17.06).  The idea being that if I am on 17.03 today and I wanted to go to 17.12, I could look at the upgrade docs for 17.06, perform the upgrade 17.03 -> 17.06, look at the docs for 17.09 and know how to upgrade from 17.06 -> 17.09 and then the same for 17.09 -> 17.12.  I'd be able to figure out how to do my incremental upgrades that way.

If we have no docs for upgrade, how is anyone supposed to upgrade knowing that it works?  It shouldn't be tribal knowledge and I do not want to be involved in every upgrade a customer has.

**Comment by ahh-docker** [comment]
Closed as stale for @mbentley
